pkg/sql/sqlliveness/slstorage/test_helpers.go

```diff
@@ -21,7 +22,8 @@ import (
 
 // FakeStorage implements the sqlliveness.Storage interface.
 type FakeStorage struct {
-	mu struct {
+	InsertSleep time.Duration
```

Comment:
Sleeps are generally an anti-pattern in our repo. Especially sleeps that are intended to wait for something to happen. Anything like this would need to be well motivated. I have a hard time understanding how these sleeps fit into the test. 

I think I'd prefer if we did something more explicit, like have a the `FakeStorage` send on a channel here or. One approach is to use a `chan chan struct{}` where the `Insert` call will send a channel on a channel, and then block until it receives from the sent channel. In that way, we can be sure that the Insert does not make progress, and we don't need to wait for timing.

Reply:
Thanks for pointing it out. Made changes and introduced the pattern you suggested.